src/openvic-simulation/map/Map.cpp

```diff
@@ -562,15 +562,26 @@ bool Map::load_province_positions(BuildingTypeManager const& building_type_manag
 	})(root);
 }
 
-bool Map::load_region_file(ast::NodeCPtr root) {
+bool Map::load_region_colours(ast::NodeCPtr root, std::vector<colour_t>& colours) {
+	return expect_dictionary(
```

Comment:
```suggestion
	return expect_dictionary_reserve_length(
		colours,
```
Currently this code will need to reallocate `colours` every time it runs out of space, this can be avoided by reserving space in the `std::vector`. `expect_dictionary_reserve_length` will reserve as much space as there are entries in the dictionary, so in this case one per colour.